### ISSUE TITLE
WIP: Open Feign must support multiple interface inheritance

### ISSUE THREAD

**Issue by silkentrance** [spring-cloud/spring-cloud-openfeign#413]
## WIP Notice
I am currently working on this feature request across multiple repositories, so this this will be updated multiple times. Once finished, this notice will be removed. As soon as this notice gets removed, there will be a PR for spring-cloud-openfeign.

## Problem

The current problem with OpenFeign is that one cannot inherit from multiple interfaces. And while the current contract,, i.e. single inheritance of annotated interfaces, should be supported in a backward compatitble way, I see no reason why one should prevent the user to inherit from plain interfaces or marker interfaces even, and recursively so.

Plain interfaces meaning that no inherited interface type nor that any of the inherited interface methodss or parameters thereof are annotated with any of the spring web framework supported annotations, such as `RequestMapping` and so on.

**Describe the solution you'd like**

**Describe alternatives you've considered**

There are no alternatives unless you come up with a better solution to the problem.

**Additional context**

I am currently working on a PR for the open feign project to incorporate such measures, meaning that the `Contract.AbstractContract` api will be extended, along with the `Contract.DefaultContract` on which this is based upon.

**Comment by OlgaMaciaszek** [comment]
Thanks, @silkentrance. We will need to see what happens with the issue and PR in Feign core first. Please comment here once the PR has been merged there.

**Comment by silkentrance** [comment]
@OlgaMaciaszek Thank you for the reply. I was already convinced that due to the ongoing pandemic, all work had been cancelled.

**Comment by OlgaMaciaszek** [comment]
No, @silkentrance we just have a small team maintaining a big number of projects. The OpenFeign team is independent from us, though, so first let's see what happens with the changes you've submitted in that repo.
